src/oidcendpoint/authn_event.py

```diff
@@ -24,30 +25,36 @@ def expires_in(self):
         return self["valid_until"] - time_sans_frac()
 
 
-def create_authn_event(uid, salt, authn_info=None, **kwargs):
+def create_authn_event(uid, authn_info=None, authn_time: int = 0,
+                       valid_until: int = 0, expires_in: int = 0, **kwargs):
     """
 
+    :param authn_time:
     :param uid:
-    :param salt:
     :param authn_info:
+    :param valid_until:
+    :param expires_in:
     :param kwargs:
     :return:
     """
 
-    args = {"uid": uid, "salt": salt, "authn_info": authn_info}
+    args = {"uid": uid, "authn_info": authn_info}
 
-    try:
-        args["authn_time"] = int(kwargs["authn_time"])
-    except KeyError:
-        try:
-            args["authn_time"] = int(kwargs["timestamp"])
-        except KeyError:
+    if authn_time:
+        args["authn_time"] = authn_time
+    else:
+        _ts = kwargs.get("timestamp")
```

Comment:
Let's add `timestamp` argument to function. We shouldn't use undocumented arguments from kwargs. This way the caller knows all the params he can pass, without having to look into the function body

Reply:
looks good, is there some exception handling if "timestamp" couldn't be available?

Reply:
OK